src/electron/electron/utils.cljs

```diff
@@ -53,16 +55,56 @@
     dirs))
 
 (defn set-fetch-agent
-  [{:keys [protocol host port] :as opts}]
-  (reset! *fetchAgent
-          (when (and protocol host port)
-            (new HttpsProxyAgent (str protocol "://" host ":" port))))
-  (cfgs/set-item! :settings/agent opts))
+  "Set proxy for fetch, and rsapi"
+  ([opts]
+   (set-fetch-agent opts true))
+  ([{:keys [protocol host port] :as opts} persist-to-config?]
+   (if (and protocol host port)
+     (let [proxy-url (str protocol "://" host ":" port)]
+       (logger/info "set proxy to" proxy-url)
+       (reset! *fetchAgent (new HttpsProxyAgent proxy-url))
+       (rsapi/setProxy proxy-url))
+     (do
+       (logger/info "reset proxy")
+       (reset! *fetchAgent nil)
+       (rsapi/setProxy nil)))
+   (when persist-to-config?
+     (cfgs/set-item! :settings/agent opts))))
+
+(defn- parse-pac-rule
+  "Parse Proxy Auto Config(PAC) line"
+  [line]
+  (let [parts (string/split line #"[ :]")
+        type (first parts)]
+    (cond
+      (= type "DIRECT")
+      nil
+
+      (and (contains? #{"PROXY" "HTTP" "SOCKS"} type)
+           (>= (count parts) 3))
+      {:protocol (if (= type "SOCKS") "socks5" "http")
+       :host (nth parts 1)
+       :port (nth parts 2)}
+
+      :else
+      (do
+        (logger/warn "Unknown PAC rule:" line)
+        nil))))
 
 (defn restore-user-fetch-agent
+  "Restore user's proxy settings"
   []
-  (when-let [agent (cfgs/get-item :settings/agent)]
-    (set-fetch-agent agent)))
+  (let [agent-opts (cfgs/get-item :settings/agent)]
+    (if (and agent-opts (not-empty (:protocol agent-opts)))
+      (set-fetch-agent agent-opts)
+      (when-let [sess (.. @*win -webContents -session)]
+        (p/let [proxy (.resolveProxy sess "https://www.google.com")
```

Comment:
The "electron-way" to acquire system proxy settings.
Need a more proper URL here, like `https://api.logseq.com` or AWS's s3 domain.